blog/_posts/2019-09-05-release-process-zh-cn.md

```diff
@@ -0,0 +1,257 @@
+
+- 大约在发布补丁版本的五天前，我们会在**反向移植**（backport）分支上运行PkgEval。
+  如果一切顺利，我们会将其**归并**（merge）并**冻结**（freeze）这些版本分支，并在discourse上宣布可以开始测试了。
+  如果在这接下来的五天里，一切风平浪静，这些版本分支会被贴上新的**版本标签**（tag）。
+
+## 次要版本（Minor releases）
+
+- 次要版本增加版本号的中间位，即**次版本号**。
+  比方说，从`1.2.3`到`1.3.0`标志着次要版本的发布。
+
+- 次要版本包含bug修复，新**特性**（feature），和一些“小改动”。
+  这些小改动理论上可能造成不兼容，但事实上很少引起不兼容。
+  更何况，我们通过PkgEval完全避免了不兼容的发生。
+
+- 次要版本也会大量地**重构**（refactor）内部代码。
+  之前提到，我们规定补丁版本只允许在修复bug的前提下**小范围**地重构内部代码，次要版本便顺理成章地成为我们**大范围**重构的工作场所。
+  如果你的程序依赖于我们的内部代码而不是公开的应用程式介面的话，你这下可能会遇到不兼容问题。
+  事实上，你之所以在补丁版本侥幸活了下来，是因为我们在补丁版本执行了比SemVer更严格的标准。
+  任何出于某种需要而依赖我们内部代码的用户，在升级次要版本时都应该格外小心。
+
+- 次要版本每**四月**发布一次，也就是每年发布三次。
+  每四个月，我们在discourse上宣布当前开发版本将在两周后冻结。
+  在冻结当天，我们为次要版本建立`release-1.3`分支（译者注：此文写作时的版本分支）。
+  该分支会被贴上版本标签，并且不允许额外添加新增特性。
+
+## 主要版本（Major releases)
+
+- 依据SemVer，主要版本可以大刀阔斧地改动。
+  不过，现实中，我们很清楚我们将如何塑造Julia的代码，并不会做出面目全非的改变。
+  大部分用户级别代码会在Julia `2.0`版本（译者注：此文写作时的未来主要版本）中完整的保留下来。
+  无理取闹地将一切规则打破并不是我们想看到的。
+
+- 主要版本的职责在于修正明显的API设计缺陷，人人都会因为能摆脱这种差劲的，扑朔迷离的API而拍手称快。
+  主要版本也允许修改底层代码，这会造成某些**第三方库**（package）的不兼容，但这是从根本上改善语言所必须付出的代价。  
+
+## 长期支持（Long term support）
+
+一些用户乐于时刻更新Julia以获得最炫最酷的新特性。
+另一些用户甚至乐此不疲地每天重新编译Julia的master分支以做第一个吃螃蟹的人。
+还有的用户，恰恰相反，一年到头也懒得升级一次。
+理想情况下，我们愿意为每个存在于世的次要版本永远提供bug修复服务。
+如果我们有无限的资源，我们会将每一个bug修复反向移植到每一个兼容的版本分支上。
+理想很丰满，现实很骨感。
+我们的资源仅够我们维护几个活跃版本分支。
+因此，我们退而求其次，决定在任何时间节点上仅维护至多四个活跃分支：
+
+- `master`分支： 这是所有新特性的发源地，大部分bug修复的栖息之处，也将在未来成为有划时代意义的`2.0`版本的摇篮。
+
+- **不稳定版本**（unstable release）分支（当前为`release-1.3`）：在这个分支上，新特性已经被冻结下来，但bug修复和性能改进仍被允许。
+  通常，bug修复先在`master`上完成并随后反向移植到该分支。
+  时机成熟后，该分支将被贴上版本标签（当前为`1.3.0`），并以新的稳定版本分支这一身份活跃。
+  不稳定版本分支并不是一直存在的：它只存在于**特性冻结**（feature freeze）后，下个次要版本前。
+  在此之后它都不会露面，直到四个月之后的下一次特性冻结。
+
+- **稳定版本**（stable release）分支（当前为`release-1.2`)：这个版本分支跟踪最新发布的次要（或主要）版本。
+  这个分支永远存在，并且通过反向移植从`master`获得一切可用的bug修复。
+  未来的补丁版本（比方说`1.2.1`）会建立在这个分支上。
+  当不稳定版本分支上位为新的稳定版本分支时，该旧有稳定版本分支将被遗弃。
+
+- **长期支持** （long term support, LTS）分支（当前为`release-1.0`)：这个稍旧的版本分支在其生命周期内将持续地获得bug修复。
+  纵使某些bug修复不能完全地反向移植到该分支，我们也会额外花力气妥善地修复该分支上的bug。
+  一个旧LTS分支将在另一个分支成为新LTS分支时退役。
+
+现在问题只剩一个：什么时候LTS分支会更替？
+`release-1.0`是我们目前唯一确立过的LTS分支。
+获得了四个补丁版本后，该分支相当稳定并被广泛支持。
+可是，它从`master`获得的bug修复补丁会与日俱减，并且越来越多的第三方库会放弃对其的支持（这些库需要用到Julia新版本中的特性）。
+当合适的时机来临时，我们不得不选择一个新的LTS分支并宣布放弃维护`1.0.x`系列。
+这个新的LTS分支可能是`1.4`或`1.8`，也可能是`2.0`。
+我们现在还无法预言，但这一天终究会到来。
+幸运的是，即便如此，`1.0.x`系列的使用者们也并非一定要升级版本。
+他们大可以使用这一旧版本，并只与和该版本兼容的第三方库版本打交道。
+到那时，它将成为最稳定，最充分测试的Julia版本。
+所以，只要你不需要新特性，你大可以放心地继续无限期地使用它。
+另外，如果某人或某组织出于自身利益，愿意继续维护某个旧版本分支，也就是**甄选**（cherry-pick）反向移植并运行PkgEval以确保兼容性，我们将很乐意接受这些帮助从而发布更多的版本。
+因此，你总可以通过自己维护或雇人维护来获得更长期的支持。
+就目前来说，`release-1.0`仍将继续是一个优秀的，稳定的LTS分支。
+并且，当我们打算更替LTS分支时，我们会提前发布大量**警告**（warning）。
+
+## 不同的版本分支对应于不同的风险承受力
+
+不同的用户有不同的**风险承受力**（risk tolerance）。
+一些风险承受力高的用户能驾轻就熟地发现并汇报零星的bug，并侦察出为什么某个第三方库与Julia的新版本不兼容。
+另一些风险承受力低的用户希望使用久经考验，日臻完善的版本。
+还有些用户介于这两个极端间的某处。
+大致可以把大多数用户根据风险承受力分为下面四类：
+
+1. **高风险承受力**（high risk tolerance）：“人生只有一次，我在master分支上翩翩起舞。况且，master分支已经不像以前那样不稳定了。现在master只偶尔抽风，就算抽风了，我也可以帮忙解决它。”
```

Comment:
这里不是很准确

> since there won’t be breaking changes on master for a while, so packages should continue to work even on master, but bugs happen, y’know?

没有提到mater现在很少有break更新这件事。

Reply:
将改为
> 况且，现在master分支在一段时间内不会有破坏兼容性的更新（译者注：根据前文，破坏兼容性的更新在开发`2.0`版本时才会出现）。现在master只偶尔出现bug，不过就算出现bug，我也可以帮忙解决。